### DISCUSSION TITLE
Is gradient colormap supported?

### DISCUSSION THREAD

**Discussion by sairahul1526** [developmentseed/titiler#781] [Q&A]
if not, is there any way to implement it?

**Comment by vincentsarago** [comment]
@sairahul1526 I'm not sure what you mean by `gradient` colormap but have you looked at https://developmentseed.org/titiler/advanced/rendering/#custom-colormaps and https://cogeotiff.github.io/rio-tiler/colormap/

**Reply by sairahul1526** [reply]
Sorry if it didn't explain clearly.

I mean it is not value based, range based custom colormap like this

Like a gradient color selection we see in Canva or most products. Like attached below.

I know this can be done with value based, like giving each pixxel on value, but that will be too much large while sending in query params. 

So wanted to know if there is way to do this

<img width="593" alt="Screenshot 2024-02-19 at 11 05 50 PM" src="https://github.com/developmentseed/titiler/assets/16268489/326d4305-ff69-428a-8cdf-255d33aa7c47">

**Reply by vincentsarago** [reply]
like https://developmentseed.org/titiler/examples/code/tiler_with_custom_colormap/

**Reply by sairahul1526** [reply]
Thanks for the reply Vicent.

But again, I was not looking for discrete colormaps, which is mentioned in the example above.

I'm looking for custom sequential colormaps, like below. If I mention some color hashs at certain pixel points, colormap should be gradient smooth like below.

Is there any param like colormap_type=sequential, to do this. Otherwise, example mentioned here gives discrete colors.

![Screenshot 2024-02-20 at 5 33 26 PM](https://github.com/developmentseed/titiler/assets/16268489/7a596b91-e3c8-4592-b54b-aacc45815814)

**Comment by yuhaihan** [comment]
I have also encountered the same problem. Have you achieved the effect of linear rendering?
